com.civicrm.hrjobroles/CRM/Hrjobroles/DAO/HrJobRoles.php

```diff
@@ -508,25 +508,27 @@ static function &fields()
         ) ,
         'hrjc_role_start_date' => array(
           'name' => 'start_date',
-          'type' => CRM_Utils_Type::T_DATE | CRM_Utils_Type::T_TIME,
+          'type' => CRM_Utils_Type::T_DATE + CRM_Utils_Type::T_TIME,
+          'default' => 'NULL',
```

Comment:
It had the default 0 before, that's why I changed it do default NULL, to avoid breaking something else. I agree with you that it should be not null, I'm just not sure it should be fixed on this ticket/PR